irc/client.go

```diff
@@ -561,6 +579,9 @@ func (session *Session) playResume() {
 	}
 
 	timestamp := session.resumeDetails.Timestamp
+	if timestamp.IsZero() {
+		timestamp = session.client.ctime
```

Comment:
I'm interested by this part here, not sure exactly what time this refers to.

If the client doesn't explicitly provide a timestamp with the `RESUME` message we should probably either keep a note of when the last message we received from the client was and use that, or just leave out any particular times

Reply:
Ah, yes. `ctime` is the time the client first connected.

I tied myself into knots a bit on the question of how to estimate when the client's connection was last known to be healthy: I ended up with something like, "take the last time we received a line from them that wasn't affected by fakelag, then subtract 15 seconds for lag/buffering". I ended up concluding that this was out of scope and left it for #524.

Basically, clients really should be sending us a timestamp, if they don't then ctime is a really safe option.

Reply:
Hmm, I reckon it makes sense for us to send the timestamp if we do that sort of tracking. For now though, we should probably just leave it as-is and not send any timestamp to the end user, since the timestamp is assumed to be roughly accurate (and if the client first connected like a week ago others shouldn't show something like "(1 week of history lost)".